### PR TITLE
promql: include parsing in active-query tracking

### PR DESCRIPTION
**Pull request by bboreham**
My primary objective here is to include parsing under the `--query.max-concurrency` limit.
The PR is structured as two refactoring commits followed by one that achieves this.

It has the side-effect of including any time queuing for that limit under `ExecQueueTime`.
I note that Prometheus does not currently report time spent parsing in its stats.

### CODE REVIEW

**Comment by roidelapluie** [comment]
I disagree with this change. The current active query tracking also accounts for consoles, rules, alerts. I would keep it strictly self-contained in the engine.

**Comment by roidelapluie** [comment]
Can't we keep the queueSpanTimer ?

**Comment by bboreham** [comment]
> Can't we keep the queueSpanTimer ?

I’m sorry, can you expand your comment?  I see in the diff three mentions of `queueSpanTimer` deleted and two added.

File: promql/engine.go [hunk 1]
```diff
@@ -623,6 +622,18 @@ func (ng *Engine) exec(ctx context.Context, q *query) (v parser.Value, ws storag
...
 
+// Log query in active log. The active log guarantees that we don't run over
+// MaxConcurrent queries.
+func (ng *Engine) queueActive(ctx context.Context, q *query) (func(), error) {
+	if ng.activeQueryTracker == nil {
+		return func() {}, nil
+	}
+	queueSpanTimer, _ := q.stats.GetSpanTimer(ctx, stats.ExecQueueTime, ng.metrics.queryQueueTime)
+	queryIndex, err := ng.activeQueryTracker.Insert(ctx, q.q)
+	queueSpanTimer.Finish()
```
Comment (roidelapluie) [inline promql/engine.go:633]: should that be in the defer?

**Review by roidelapluie — APPROVED** [review]
LGTM

**Pull request merged**